src/PrestaShopBundle/ApiPlatform/Provider/QueryProvider.php

```diff
@@ -28,33 +28,105 @@
 
 namespace PrestaShopBundle\ApiPlatform\Provider;
 
+use ApiPlatform\Metadata\CollectionOperationInterface;
 use ApiPlatform\Metadata\Operation;
 use ApiPlatform\State\ProviderInterface;
 use PrestaShop\PrestaShop\Core\CommandBus\CommandBusInterface;
+use PrestaShopBundle\ApiPlatform\Converters\ConverterInterface;
 use PrestaShopBundle\ApiPlatform\Exception\NoExtraPropertiesFoundException;
+use Symfony\Component\Serializer\Exception\ExceptionInterface;
 use Symfony\Component\Serializer\Serializer;
 
 class QueryProvider implements ProviderInterface
 {
     public function __construct(
         private readonly CommandBusInterface $queryBus,
+        private readonly iterable $converters,
         private readonly Serializer $apiPlatformSerializer
     ) {
     }
 
+    /**
+     * @throws \ReflectionException
+     * @throws \Exception
+     * @throws ExceptionInterface
+     */
     public function provide(Operation $operation, array $uriVariables = [], array $context = [])
     {
-        $extraProperties = $operation->getExtraProperties();
+        $queryClass = $operation->getExtraProperties()['query'] ?? null;
+        $filters = $context['filters'] ?? [];
```

Comment:
What does `$context['filters']` refer to? Is it the parameters provided via form post? Or are they the extra GET parameters?

Reply:
It's extra GET parameters ;)